hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSUtilClient.java

```diff
@@ -661,9 +661,12 @@ public static boolean isValidName(String src) {
     String[] components = StringUtils.split(src, '/');
     for (int i = 0; i < components.length; i++) {
       String element = components[i];
+      // For Windows, we must allow the : in the drive letter.
+      if (Shell.WINDOWS && i == 1 && element.contains(":")) {
```

Comment:
Could you please make an `endsWith` check for `:` instead of `element.contains`?

Reply:
OK, Thanks